IPs/RSKIP517.md

```diff
@@ -0,0 +1,144 @@
+
+## Specification
+
+### Parameters and definitions
+
+* $B$ = last N blocks
+* $B_i$ = block in position $i$ of $B$, where $B_1$ is the first block in the set and $B_N$ is the last block.
+* $T_{B_i}$ = block time for block $i$
+* $U_{B_i}$ = uncles referenced by block $i$
+* $D_{B_i}$ = difficulty of block $i$
+* $T$ = target time between blocks in seconds (currently is set to 14)
+* $C$ = uncle threshold (e.g., 0.5 means threshold supports up to 1 uncle every 2 blocks)
+* $\alpha$ = base difficulty change factor (currently is set to 0.0025)
+
+### Background
+
+Each time a new block is mined, the difficulty for the next block is recalculated using the most recent block's difficulty and a scaling factor $F$. This adjustment helps maintain a consistent block interval by increasing or decreasing difficulty in response to recent network conditions. The formula used is:
+
+$$
+D_{B_{N+1}} = D_{B_{N}} \cdot (1 + F)
+$$
+
+The current difficulty adjustment factor can be defined as:
+
+$$
+F = \begin{cases}
+\alpha & \text{if }  T\cdot (1+U_{B_N}) \geq T_{B_N} \\
+-\alpha & \text{if } T\cdot (1+U_{B_N}) < T_{B_N} \\
+\end{cases}
+$$
+
+RSKIP77 proposes an alternative method that uses an N-block (N=32) sliding window:
+
+$$
+F = \begin{cases}
+\alpha & \text{if } \left(\sum_{i=1}^{N} 1 + U_{B_i}\right)\cdot T \geq T_{B_N} - T_{B_1} \\
+-\alpha & \text{if } \left(\sum_{i=1}^{N} 1 + U_{B_i}\right)\cdot T < T_{B_N} - T_{B_1} \\
+\end{cases}
+$$
+
+While RSKIP77¹ smooths difficulty oscillations by broadening the analysis window, it still incorporates uncle blocks directly into the calculation. This proposal takes a step further by separating the treatment of uncles and blocks, introducing a threshold mechanism.
+
+### New difficulty calculation
+
+We define the average block time $A$ as:
+
+$$
+A = \frac{\sum_{i=1}^{N} T_{B_i}}{N}
+$$
+
+and the uncle rate $R$ as:
+
+$$
+R = \frac{\sum_{i=1}^{N} U_{B_i}}{N}
+$$
+
+The new difficulty adjustment introduces a conditional factor $F$ that adapts based on both the uncle rate and deviation from the target block time. This mechanism ensures that uncle blocks impact difficulty when their rate exceeds a defined threshold, preserving network stability while allowing controlled block time reductions. The adjustment factor is defined as:
+
+$$
+F = \begin{cases}
+\alpha & \text{if } R > C \\
+\alpha & \text{if } R \leq C \land A > T \cdot 1.10 \\
+-\alpha & \text{if } R  \leq C  \land A < T \cdot 0.90 \\
+0 & \text{otherwise}
+\end{cases}
+$$
```

Comment:
The actual computation should be specified because depending on how this is computed (bigint arithmetic, float arithmetic, etc.) it may give different results.

I suggest to add. here the exact variable types that will be used and analyze the performance of those computations. Maybe computing the coheficients using fixed-point numbers (i.e. 2<<10 multiplier) in uint64 variables can be more efficient.